### ISSUE TITLE
paymentMethodRequestable/paymentMethodIsSelected with 3DS

### ISSUE THREAD

**Issue by ae-michal-mordarski** [braintree/braintree-web-drop-in#805]
### General information

* SDK/Library version: 1.33.0
* Environment: Sandbox
* Browser and OS: Latest version of Chrome/Windows 10 Pro

### Issue description

Per documentation paymentMethodIsSelected will be true any time a payment method is visibly selected in the Drop-in UI, however with 3DS enabled you get false first as expected, followed by true right away, but 3DS pop-up is shown, so technically speaking payment  method is not visibly selected until 3DS is completed. 

Furthermore, we have another logic that depends on paymentMethodIsSelected being true when saved credit card is selected, thus making it impossible to distinguish, because true also comes while 3DS pop-up is visible.

https://braintree.github.io/braintree-web-drop-in/docs/current/Dropin.html#~paymentMethodRequestablePayload

1. Enter credit cards details which requires 3DS
2. requestPaymentMethod (3DS pop-up is shown and card is not selected)

Expected result: paymentMethodIsSelected is true when 3DS is completed
Actual result: paymentMethodIsSelected is true when 3DS popup is shown

```
instance.on('paymentMethodRequestable', function (event) {
	if (!event.paymentMethodIsSelected) {
		console.log('requestPaymentMethod')
		instance.requestPaymentMethod(
			{
				threeDSecure: getThreeDSecure(),
			},
			function (err, payload) {
				if (err) {
					instance.clearSelectedPaymentMethod();
					return;
				}

				console.log(payload);
			},
		);
	}
	else
	{
		console.log('paymentMethodIsSelected')	
	}
});
```

**Comment by jplukarski** [comment]
Hey @ae-michal-mordarski , thank you for reporting this and your patience while we look into this. We agree that it makes sense for the `paymentMethodRequestable` event to not report `paymentMethodIsSelected` as `true` until the 3DS challenge is completed. We have this task on our radar and will update this issue once the fix is in place.

**Comment by ae-michal-mordarski** [comment]
@jplukarski nice, I'm glad I'm able to help this way.

**Comment by jplukarski** [comment]
This bug has been fixed in version 1.42.0.
